File: 프로그래머스/1/176963. 추억 점수/추억 점수.js
function solution(name, yearning, photo) {
    var answer = [];
    for(let j = 0; j<photo.length; j++){ // 사진마다
        let miniSum = 0;
        for(let k = 0; k<photo[j].length; k++){ // 인물마다
            for(let i = 0; i<name.length; i++){ // 이름 찾기
                if(name[i] === photo[j][k]) {
                    miniSum += yearning[i];
                    break;
                }
            }       
        }
        answer[j] = miniSum;
    }
    return answer;
}

File: 프로그래머스/1/176963. 추억 점수/추억 점수.java
import java.util.*;

class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        ArrayList<Integer> result = new ArrayList<>();
        
        HashMap<String, Integer> friends = new HashMap<>();
        for(int i = 0; i<name.length; i++){
            friends.put(name[i], yearning[i]);
        } // HashMap Setting 
        
        for(String[] to : photo){
            int smallResult = 0;
            for(String pho : to) {
                // includes를 쓰지 말고
                try {
                    smallResult += friends.get(pho);
                } catch (Exception e) {
                }
            }
            result.add(smallResult);
        }
        
        int[] answer = new int[photo.length];
        int cursor = 0;
        for(Integer r : result){
            answer[cursor] = r;
            cursor++;
        }
        return answer;
    }
}